src/Compilers/VisualBasic/Portable/Symbols/NamedTypeSymbol.vb

```diff
@@ -1150,7 +1150,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             Dim baseAllRequiredMembers = If(BaseTypeNoUseSiteDiagnostics?.AllRequiredMembers, ImmutableSegmentedDictionary(Of String, Symbol).Empty)
             Dim typeHasDeclaredRequiredMembers = HasAnyDeclaredRequiredMembers
 
-            If (Not typeHasDeclaredRequiredMembers) AndAlso baseAllRequiredMembers.IsEmpty Then
```

Comment:
>If (Not typeHasDeclaredRequiredMembers) AndAlso baseAllRequiredMembers.IsEmpty Then [](http://example.com/codeflow?start=12&length=83)

I would prefer removal of this condition to be accompanied with a unit-test that observes the behavior change. Ideally, in a separate PR, not tied to a perf optimization.

Reply:
➡️ The original fast-path check has now been restored. I'll file an issue related to this check.